packages/binding-http/src/http-server.ts

```diff
@@ -683,11 +691,24 @@ export default class HttpServer implements ProtocolServer {
                             if (req.method === "GET") {
                                 thing
                                     .readAllProperties()
-                                    .then((value: any) => {
-                                        const content = ContentSerdes.get().valueToContent(value, undefined); // contentType handling? <any>property);
-                                        res.setHeader("Content-Type", content.type);
+                                    .then(async (propMap: WoT.PropertyReadMap) => {
+                                        // Note we create one object to return, TODO piping response
+                                        let recordReponse: Record<string, any> = {};
+                                        for (let key of propMap.keys()) {
+                                            let value: WoT.InteractionOutput = propMap.get(key);
+                                            value.form = { f: "all" }; // to avoid missing form error
+                                            let content = ContentSerdes.get().valueToContent(
+                                                value.data && !value.dataUsed ? value.data : await value.value(),
+                                                undefined,
+                                                "application/json"
+                                            );
+                                            // TODO ProtocolHelpers.readStreamFully() does not work for counter example for SVG countAsImage
+                                            const data = await ProtocolHelpers.readStreamFully(content.body);
+                                            recordReponse[key] = data.toString(); // contentType handling?
+                                        }
+                                        res.setHeader("Content-Type", "application/json"); // contentType handling?
```

Comment:
read all properties is *very* messy... piping does not work for me.

Moreover, `ProtocolHelpers.readStreamFully()` does not worj for all properties, e.g. SVG image in counter

Reply:
I agree it is messy, but I think this is something from the spec itself. In all the aggregated operations we are failing to describe how the payload will look like... I remembered that we decided to go with the fact that each property value should be transmitted inside a JSON object similar to what you're doing. However, don't you think is better to just call the `value` function on each `key` in `propMap` and If it fails read the stream? Why are you using `valueToContent` ?